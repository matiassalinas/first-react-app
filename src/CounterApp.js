// 'rafcp' + tab (visual studio code)
import React, { useState} from 'react'

import PropTypes from 'prop-types'

const CounterApp = ( { value = 10 } ) => {

    const [ counter, setCounter ] = useState( value );
    // Handle counter-button
    const handleCounterButtonAdd = (e) => {
        //setCounter( (c) => c + 1 )
        setCounter(counter + 1);
    }

    const handleCounterButtonSubtract = () => setCounter( counter - 1);
    const handleCounterButtonReset = () => setCounter( value );

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <button className="counter-button" onClick={ handleCounterButtonAdd }>+1</button>
            <button className="counter-button" onClick={ handleCounterButtonReset }>Reset</button>
            <button className="counter-button" onClick={ handleCounterButtonSubtract }>-1</button>
        </>
    );
}

CounterApp.propTypes = {
    value : PropTypes.number
}

export default CounterApp;
