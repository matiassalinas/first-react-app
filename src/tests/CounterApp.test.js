import React from 'react';
import { shallow } from 'enzyme'

import CounterApp from '../CounterApp';

/*
* Testing using enzyme + jest
* This testing uses snapshots. Be care reloading snapshots 
*/

describe('Testing <CounterApp /> App Component', () => {
    
    let wrapper = shallow( <CounterApp /> ); // initialization for code recomendation (vscode)
    // initialization before each test
    beforeEach( () =>  {
        wrapper = shallow( <CounterApp /> );
    });

    test('Load <CounterApp /> correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('Show default value correctly', () => {
        const default_value = 50;
        const wrapper = shallow( <CounterApp value={ default_value } /> );
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe(`${default_value}`);
    })

    test('Increase value by 1', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe('11');
    })

    test('Decrease value by 1', () => {
        // using beforeEach to setup the test (without re-initialization, it would return 10)
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe('9');
    });


    test('reset default_value', () => {
        const default_value = 50;
        const wrapper = shallow( <CounterApp value={ default_value } /> );
        // Upload value
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        // Reset
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe(`${default_value}`);
    })
})
