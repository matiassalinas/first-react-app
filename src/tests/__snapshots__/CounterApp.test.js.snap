// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CounterApp /> App Component Load <CounterApp /> correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    className="counter-button"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="counter-button"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="counter-button"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
